### PR TITLE
BUG: na parameter for str.startswith and str.endswith not propagating for Series with categorical dtype

### LINKED ISSUE #36241: BUG: .str.startswith(..., na=False) consistency between categorical and string series (again)
#### Code Sample, a copy-pastable example

```pycon
>>> import pandas as pd
>>> import numpy as np
>>> s = pd.Series(["a", "b", np.nan], index=["a", "b", np.nan], dtype="category")
>>> df = pd.DataFrame().assign(
...     cat_contains=s.str.contains("a", na=False),
...     cat_startswith=s.str.startswith("a", na=False),
...     cat_endswith=s.str.endswith("a", na=False),
...     str_contains=s.astype("string").str.contains("a", na=False),
...     str_startswith=s.astype("string").str.startswith("a", na=False),
...     str_endswith=s.astype("string").str.endswith("a", na=False),
... )
>>> df.info()
<class 'pandas.core.frame.DataFrame'>
Index: 3 entries, a to nan
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   cat_contains    3 non-null      bool
 1   cat_startswith  2 non-null      object
 2   cat_endswith    2 non-null      object
 3   str_contains    3 non-null      boolean
 4   str_startswith  3 non-null      boolean
 5   str_endswith    3 non-null      boolean
dtypes: bool(1), boolean(3), object(2)
memory usage: 93.0+ bytes
>>> df
     cat_contains cat_startswith cat_endswith  str_contains  str_startswith  str_endswith
a            True           True         True          True            True          True
b           False          False        False         False           False         False
NaN         False            NaN          NaN         False           False         False
```

#### Problem description

`.str.startswith(..., na=False)` and `.str.endswith` should make missing values `False` when the calling series is of type categorical just like it does for string series.

Similar to #22158, but `.str.contains` works here.

#### Expected Output

```pycon
>>> df
     cat_contains cat_startswith cat_endswith  str_contains  str_startswith  str_endswith
a            True           True         True          True            True          True
b           False          False        False         False           False         False
NaN         False          False        False         False           False         False
```

#### Output of ``pd.show_versions()``

Using conda env with `conda create -n pandas112 -c conda-forge pandas=1.1.2`

<details>

INSTALLED VERSIONS
------------------
commit           : 2a7d3326dee660824a8433ffd01065f8ac37f7d6
python           : 3.8.5.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 10, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : Korean_Korea.949

pandas           : 1.1.2
numpy            : 1.19.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2.3
setuptools       : 49.6.0.post20200814
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by asishm**
- [x] closes #36241
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

I parametrized dtype in existing tests. Should I create a separate test instead?

### CODE REVIEW

**Review by dsaxton — COMMENTED** [review]
Thanks @asishm, can you add a 1.1.3 release note?

File: pandas/tests/test_strings.py [hunk 1]
```diff
@@ -838,8 +838,12 @@ def test_contains_for_object_category(self):
         expected = Series([True, False, False, True, False])
         tm.assert_series_equal(result, expected)
 
-    def test_startswith(self):
-        values = Series(["om", np.nan, "foo_nom", "nom", "bar_foo", np.nan, "foo"])
+    # add category dtype parametrizations for GH-36241
+    @pytest.mark.parametrize("dtype", [None, "category"])
+    def test_startswith(self, dtype):
+        values = Series(
+            ["om", np.nan, "foo_nom", "nom", "bar_foo", np.nan, "foo"], dtype=dtype
```
Comment (jreback) [inline pandas/tests/test_strings.py:845]: also pls tests with pd.NA as well (you can parameterize over the null value as well

File: pandas/tests/test_strings.py [hunk 2]
```diff
@@ -838,8 +838,12 @@ def test_contains_for_object_category(self):
         expected = Series([True, False, False, True, False])
         tm.assert_series_equal(result, expected)
 
-    def test_startswith(self):
-        values = Series(["om", np.nan, "foo_nom", "nom", "bar_foo", np.nan, "foo"])
+    # add category dtype parametrizations for GH-36241
+    @pytest.mark.parametrize("dtype", [None, "category"])
```
Comment (jreback) [inline pandas/tests/test_strings.py:842]: we want to parameterize over 'string' dtype as well rigth?

File: pandas/tests/test_strings.py [hunk 3]
```diff
@@ -867,8 +871,12 @@ def test_startswith(self):
         )
         tm.assert_series_equal(rs, xp)
 
-    def test_endswith(self):
-        values = Series(["om", np.nan, "foo_nom", "nom", "bar_foo", np.nan, "foo"])
+    # add category dtype parametrizations for GH-36241
+    @pytest.mark.parametrize("dtype", [None, "category"])
```
Comment (jreback) [inline pandas/tests/test_strings.py:875]: same as above

File: pandas/tests/test_strings.py [hunk 4]
```diff
@@ -867,8 +871,12 @@ def test_startswith(self):
         )
         tm.assert_series_equal(rs, xp)
 
-    def test_endswith(self):
-        values = Series(["om", np.nan, "foo_nom", "nom", "bar_foo", np.nan, "foo"])
+    # add category dtype parametrizations for GH-36241
```
Comment (jreback) [inline pandas/tests/test_strings.py:874]: add inside the test

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_strings.py (hunk 2 above)
Comment (asishm) [reply pandas/tests/test_strings.py:842]: string arrays are already tested here
https://github.com/pandas-dev/pandas/blob/1.1.x/pandas/tests/test_strings.py#L3530-L3531

parametrizing over string dtype and na=True/False was making it a bit tricky as these methods return boolean series causing a dtype mismatch (boolean vs object)

Comment (asishm) [reply pandas/tests/test_strings.py:845]: @jreback tried to do that changing the referenced tests slightly with
https://github.com/pandas-dev/pandas/pull/36249/files#diff-683f8dacd08abda4913680fafe3a4ea7R3573-R3576 , https://github.com/pandas-dev/pandas/pull/36249/files#diff-683f8dacd08abda4913680fafe3a4ea7R32-R33 and https://github.com/pandas-dev/pandas/pull/36249/files#diff-683f8dacd08abda4913680fafe3a4ea7R63-R64

File: pandas/tests/test_strings.py [hunk 5]
```diff
@@ -838,8 +838,12 @@ def test_contains_for_object_category(self):
         expected = Series([True, False, False, True, False])
         tm.assert_series_equal(result, expected)
 
-    def test_startswith(self):
-        values = Series(["om", np.nan, "foo_nom", "nom", "bar_foo", np.nan, "foo"])
+    # add category dtype parametrizations for GH-36241
+    @pytest.mark.parametrize("dtype", [None, "category"])
+    def test_startswith(self, dtype):
```
[the remark this answers is not in the document]
Comment (asishm) [reply pandas/tests/test_strings.py]: parameterized over True, False

question here:

for string Series with dtype = object, passing strings into the `na` parameter of these two methods causes it to replace the nan with the string - but for StringArrays it replaces the nans with `bool(na)`. The current fix behaves as the former. 
```py
In [49]: pd.Series(['a', np.nan, 'b']).str.startswith('a', na='c')
Out[49]:
0     True
1        c
2    False
dtype: object

In [50]: pd.Series(['a', np.nan, 'b'], dtype='string').str.startswith('a', na='c')
Out[50]:
0     True
1     True
2    False
dtype: boolean

In [51]: pd.Series(['a', np.nan, 'b'], dtype='string').str.startswith('a', na='c')
Out[51]: 
0     True
1     True
2    False
dtype: boolean
```

File: doc/source/whatsnew/v1.1.3.rst [hunk 6]
```diff
@@ -22,7 +22,7 @@ Fixed regressions
 
 Bug fixes
 ~~~~~~~~~
--
+- Bug in :meth:`str.startswith` and :meth:`str.endswith` for Series with ``category`` dtype not propagating ``na`` parameter (:issue:`36241`)
```
Comment (dsaxton) [inline doc/source/whatsnew/v1.1.3.rst:25]:
```suggestion
- Bug in :meth:`Series.str.startswith` and :meth:`Series.str.endswith` with ``category`` dtype not propagating ``na`` parameter (:issue:`36241`)
```

File: pandas/tests/test_strings.py [hunk 7]
```diff
@@ -838,15 +842,22 @@ def test_contains_for_object_category(self):
...
+        )
 
         result = values.str.startswith("foo")
         exp = Series([False, np.nan, True, False, False, np.nan, True])
         tm.assert_series_equal(result, exp)
 
-        result = values.str.startswith("foo", na=True)
-        tm.assert_series_equal(result, exp.fillna(True).astype(bool))
+        result = values.str.startswith("foo", na=na)
+        tm.assert_series_equal(result, exp.fillna(na).astype(bool))
```
Comment (dsaxton) [inline pandas/tests/test_strings.py:860]: Could instead use `na` when constructing `exp` and not have to fill / astype. (Also a bit nitty but I'd spell out expected as in most other tests.)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @asishm

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**